src/events/tab_event_handler.ts

```diff
@@ -0,0 +1,82 @@
+import yaml from "js-yaml";
+import vscode from 'vscode';
+import path from 'path';
+import fs from 'fs';
+
+import { VSCODE } from "../constants";
+import { ITab, ITabFile } from "../interface";
+import { formatDatetime, getBranch } from "../utils/common";
+import { ConfigUtils } from "../utils/config_utils";
+import { pathUtils } from "../utils/path_utils";
+import { generateSettings } from "../settings";
+import { RepoState } from "../utils/repo_state_utils";
+import { UserState } from "../utils/user_utils";
+
+export class tabEventHandler {
+	repoPath = "";
+	branch = "";
+	pathUtils: any;
+
+	// Diff props
+	settings = generateSettings();
+	shouldProceed = false;
+
+	constructor(repoPath="") {
+		const userState = new UserState();
+		const isValidAccount = userState.isValidAccount();
+		this.repoPath = repoPath || pathUtils.getRootPath();
+		if(!this.repoPath) return;
+		const repoState = new RepoState(this.repoPath).get();
+		const repoIsConnected = repoState.IS_CONNECTED;
+		this.shouldProceed = isValidAccount && repoIsConnected;
+		if (!this.shouldProceed) return;
+		this.branch = getBranch(this.repoPath);
+		this.pathUtils = new pathUtils(this.repoPath, this.branch);
+	}
+
+	handleTabChangeEvent = (isTabEvent: boolean = true) => {
+		if(!this.repoPath) return;
+		// Discard event if file is changed 
+		if (!isTabEvent) return;
+		// Record timestamp
+		const created_at = formatDatetime(new Date().getTime());
+		// For the current open repoPath, get the repo_id and from config File
+		const configUtils = new ConfigUtils();
+		const repoId = configUtils.getRepoIdByPath(this.repoPath);
+		if (!repoId) return
+		const configJSON = configUtils.config;
+
+		// Get list of current tabs
+		const open_tabs = vscode.window.tabGroups.all;
+		const tabs: ITabFile[] = [];
+		// Loop through tab groups
+		for (const tab_group of open_tabs) {
+			for (let tab of tab_group.tabs) {
+				// Get path of tab
+				// @ts-ignore
+				const fileRelativePath = (tab.input.uri.path).split(`${this.repoPath}${path.sep}`)[1];
```

Comment:
We can do this inside `getFileIdByPath` since we have both filePath and repoPath in that method, we'll first get relativePath and then its ID